include/openthread/types.h

```diff
@@ -683,6 +683,8 @@ enum
     OT_CHANGED_THREAD_CHILD_REMOVED         = 1 << 11,  ///< Child was removed
     OT_CHANGED_IP6_MULTICAST_SUBSRCRIBED    = 1 << 12,  ///< Subscribed to a IPv6 multicast address
     OT_CHANGED_IP6_MULTICAST_UNSUBSRCRIBED  = 1 << 13,  ///< Unsubscribed from a IPv6 multicast address
+    OT_CPU_POWER_HI                         = 1 << 14,  ///< Hi CPU power is needed
```

Comment:
This PR is part of performance and battery consumption optimization for DA15000. Correct me if I am wrong, but most of other examples does not fully cover that topic. I have seen examples providing support by HW crypto functions, which will be also introduce in our next upcoming PR, but this particular PR is rather connected to battery consumption topic. You might see this topic as increasing CPU clock for DTLS operations, but our original idea was to safe battery by decreasing CPU clock frequency everywhere it is possible. From that perspective we do not treat it as Commissioner and Joiner issue. Those are cases were high performance is mandatory and due to that cannot or we just do not see a point to take care about battery devices. 

Commissioner with battery is rather not good idea, so that why we do not care so much about it and we do not need to slow down CPU for that case. In case of SED device which will need to act for a short time as JOINER it is much important to limit CPU usage as much as possible.

I would agree that CPU_HIGH_POWER might be not the best names, but  names OT_CHANGED_COMMISSIONER_STATE or OT_CHANGED JOINER_STATE will hide the overall idea of this PR. It would be great if other platform could utilize this approach and optimize battery consumption by that. In the worst scenario this change on OpenThread code would be done just for DA15000 purpose.

What do you think about HIGH_PERFORMANCE_NEEDED and BATTERY_SAVING_POSSIBLE states? Someone could complain that, those might not representing real states, but it will shows the idea of this PR.